File: Java/Java source-code/autoCorrelation.java


public class autoCorrelation {
    public static double autoCorrelation_m(Double[] data, Integer lag, Double mean, Double variance) {
        int N = data.length;
        double run = 0;
        for (int i = lag; i < N; ++i) {
            run += (data[i] - mean) * (data[i - lag] - mean);
        }
        return run / (N - lag) / variance;
    }
}

File: C++/autoCorrelation.cpp
#include <iostream>

double autoCorrelation(double data[], int lag, double mean, double variance, int n){
    double run = 0;
    for (int i = lag; i < n; i++)
    {
        run += (data[i] - mean) * (data[i - lag] - mean);
    }
    
    return run / (n - lag) / variance;
}